narwhal/primary/src/primary.rs

```diff
@@ -550,6 +541,221 @@ impl PrimaryReceiverHandler {
         }
         Ok(None)
     }
+
+    async fn process_request_vote(
+        &self,
+        request: anemo::Request<RequestVoteRequest>,
+    ) -> DagResult<RequestVoteResponse> {
+        let network = request
+            .extensions()
+            .get::<anemo::NetworkRef>()
+            .and_then(anemo::NetworkRef::upgrade)
+            .ok_or_else(|| {
+                DagError::NetworkError("Unable to access network to send child RPCs".to_owned())
+            })?;
+
+        let header = &request.body().header;
+        let committee = self.committee.load();
+        header.verify(&committee, self.worker_cache.clone())?;
+
+        // Vote request must come from the Header's author.
+        // DO NOT MERGE: Will enforcing RPC requester is same as Header author cause any issues?
```

Comment:
Discussed on Zoom but for posterity: this is required for thread safety with the `vote_digest_store` because otherwise multiple vote requests from the same author could race. It's also good protection against byzantine nodes, who can now only cause at most 1 inflight RPC worth of bogus batch syncs.